### PR TITLE
Code upload

### PR DESCRIPTION
**Pull request by abhikedia**
test.sh not working.

### CODE REVIEW

File: Assignments/Splitwise/CPP/main.cpp [hunk 1]
```diff
@@ -0,0 +1,264 @@
+#include <bits/stdc++.h>
+using namespace std;
+class Users
+{
+public:
+    map<string, map<string, float>> users;
+    void addUsers(vector<string>);
+    void getUser(string);
+    void getUser();
+    void addExpense(string, string, string, vector<string>);
```
Comment (sribs) [inline Assignments/Splitwise/CPP/main.cpp]: This code is not extensible. Let me pose this question. 
Let us say I wanted to add another expense of a different type. How will you achieve this ?

File: test.sh [hunk 2]
```diff
@@ -3,20 +3,22 @@ set -e
...
-fi
+# else
+#     cd $WD
+#     cat ../input.txt | python3 main.py > test.txt
+ #   cd ../..
+#fi
 
 echo "==================Testing=================="
 diff $WD/test.txt $assignment/output.txt > $WD/final
+echo "Crossed"
```
Comment (sribs) [inline test.sh]: Can you output the Results in ``` README.md ```

**Review by sribs — CHANGES_REQUESTED** [review]

File: Assignments/Splitwise/CPP/main.cpp [hunk 3]
```diff
@@ -7,9 +7,9 @@ class Users
     void addUsers(vector<string>);
     void getUser(string);
     void getUser();
-    void addExpense(string, string, string, vector<string>);
-    void addExpense(string, vector<string>, vector<string>);
-    void addExpense(string, vector<string>, vector<string>, string);
+    void addExpense(string, string, string, vector<string>);  //for EQUAL
```
Comment (sribs) [inline Assignments/Splitwise/CPP/main.cpp]: Well, this is still not what I'm expecting. Think of how you would use this app and how would you design. How you as a developer add incoming features in future versions. Would you end up overloading 100's of Add expense methods

**Review by sribs — CHANGES_REQUESTED** [review]

File: Assignments/Splitwise/CPP/main.cpp (hunk 3 above)
Comment (abhikedia) [reply Assignments/Splitwise/CPP/main.cpp]: Removed multiple addExpense function. And split the code in multiple files.

File: Assignments/Splitwise/CPP/function_definitions.cpp [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+            {
+                if (this->users[a[0]][receivers[i]] >= amount / receivers.size())
+                {
+                    this->users[a[0]][receivers[i]] -= amount / receivers.size();
+                    continue;
+                }
+                this->users[receivers[i]][a[0]] += amount / receivers.size();
+            }
+        }
+    }
```
Comment (sribs) [inline Assignments/Splitwise/CPP/function_definitions.cpp:92]: I Stilll dont get it. It is still not a maintainable code. Please dont make me to tell the answer. You are not using principles of oops

File: Assignments/Splitwise/CPP/function_definitions.cpp [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        vector<float> balance;
+        float temp;
+        for (int i = 2; i < a.size() - 1; i++)
+        {
+            stringstream change(a[i]);
+            change >> temp;
+            sum += temp;
+            balance.push_back(temp);
+        }
+        if (a[a.size() - 1] == "EXACT")
```
Comment (sribs) [inline Assignments/Splitwise/CPP/function_definitions.cpp:104]: I Stilll dont get it. It is still not a maintainable code. Please dont make me to tell the answer. You are not using principles of oops

**Review by sribs — CHANGES_REQUESTED** [review]

File: Assignments/Splitwise/CPP/main.cpp (hunk 3 above)
Comment (sribs) [reply Assignments/Splitwise/CPP/main.cpp]: No that is not a viable solution either

**Comment by sribs** [comment]
Can we have an update ?

**Comment by abhikedia** [comment]
Working on Trello assignment. My Current solution is very inefficient. Trying to improve it.

**Pull request closed without merging**